frontends/election-manager/src/utils/election.ts

```diff
@@ -264,3 +264,45 @@ export function generateBlankBallots({
 
   return ballots;
 }
+
+export function generateOvervoteBallot({
+  election,
+  precinctId,
+}: {
+  election: Election;
+  precinctId: PrecinctId;
+}): Dictionary<string | VotesDict> | undefined {
+  const precinctBallotStyles = election.ballotStyles.filter((bs) =>
+    bs.precincts.includes(precinctId)
+  );
+
+  const ballot: Dictionary<string | VotesDict> = { precinctId };
+  const votes: VotesDict = {};
+
+  for (const ballotStyle of precinctBallotStyles) {
+    const contests = election.contests.filter(
+      (c) =>
+        ballotStyle.districts.includes(c.districtId) &&
+        ballotStyle.partyId === c.partyId
+    );
+
+    for (const contest of contests) {
+      if (contest.type === 'yesno') {
+        votes[contest.id] = ['yes', 'no'];
+      } else if (contest.type === 'ms-either-neither') {
+        votes[contest.eitherNeitherContestId] = ['yes', 'no'];
+      } else if (
+        contest.type === 'candidate' &&
+        contest.candidates.length > contest.seats
+      ) {
+        votes[contest.id] = contest.candidates.slice(0, contest.seats + 1);
+      } else {
+        continue;
+      }
+
+      ballot.ballotStyleId = ballotStyle.id;
+      ballot.votes = votes;
+      return ballot;
+    }
+  }
```

Comment:
> Updated snapshot. @arsalansufi it's worth noting that the snapshots used for HMPB ballots are mocks that include the precinct, election, and ballot style, but don't actually include the votes. Same as in https://github.com/votingworks/vxsuite/pull/1808. So we are testing that an additional ballot of the correct ballot style is being added, but not that it is properly marked.

Ooh good to know! Let's test this function through additional means then. A set of unit tests targeted at this helper, covering the various contest cases as well as edge cases (no contest to validly overvote, etc.), could work well!

In some of our other frontend packages (frontends/bmd), we have automated code coverage checks that verify that every branch of code is covered by a test. While we don't have that set up in frontends/election-manager, I still think the "cover every branch" mindset is helpful, in this case writing tests that cover all the branches within this helper